source/d/gc/arena.d

```diff
@@ -43,18 +46,23 @@ private:
 		return filler.usedPages;
 	}
 
+	private static shared void* _arenaStore;
```

Comment:
Why can't this be typed as `shared(Arena*)` ?

Reply:
I tested, and it could be this. But I dislike not specifying the storage class as shared (vs. just the type). It leaves doubts (is this a TLS-stored pointer that can be shared?).

Is there a reason to prefer specifying shared via the type constructor instead of storage class?

Reply:
My comment was mostly about going from untyped (`void*`) to typed (`Arena*`). Using `shared(X*)` or `shared X*` should be exactly the same, so I do not care which we use.

Reply:
Oh, I didn't notice the difference! Yes, it needs to be `void*`, because the arenas are cache-aligned (see how `ArenaSize` is not `Arena.sizeof`)

That is, arena element at index `i` is not `(cast(Arena*)_arenaStore)[i]`